src/main/java/DeleteCommand.java

```diff
@@ -0,0 +1,29 @@
+import java.util.ArrayList;
+
+class DeleteCommand extends Command {
+    int index;
+
+    public DeleteCommand(int index) {
+        this.index = index;
+    }
+
+    public boolean terminate() {
```

Comment:
Instead of having terminate() return false for all commands except ExitCommand, I had the ExitCommand initialised to set terminate (a boolean in Command abstract class) to true. 
Hence only the ExitCommand class will refer to terminate().

Reply:
I originally envisioned `Command` as an interface, so that was how it was written at the start. However, I agree that your suggestion would make my code cleaner and will implement it.